index.js

```diff
@@ -0,0 +1,265 @@
+        } else {
+            this.bindEvent( key, this.events[key].event )
+        }
+    }
+
+    async delete( { silent } = { silent: false } ) {
+        await this.hide()
+
+        const container = this.els.container,
+            parent = container.parentNode
+        if( container && parent ) parent.removeChild( container )
+        if( !silent ) this.emit('deleted')
+    }
+
+    events = { }
+
+    fadeInImage( el ) {
+        el.onload = () => {
+            this.emit( 'imgLoaded', el )
+            el.removeAttribute('data-src')
+        }
+
+        el.setAttribute( 'src', el.getAttribute('data-src') )
+    }
+
+    getEventMethodName( key, event ) { return `on${this.capitalizeFirstLetter(key)}${this.capitalizeFirstLetter(event)}` }
+
+    getContainer() { return this.els.container }
+
+    getTemplateOptions() {
+       return { user: this.user ? this.user.data : {}, model: this.model }
+    }
+
+    handleLogin() {
+        this.factory.create( 'login', { insertion: { el: document.querySelector('#content') } } )
+        .on( "loggedIn", () => this.onLogin() )
+
+        return this
+    }
+
+    async hide( isSlow ) {
+        if( !this.els || this.isHiding ) return
+
+        this.isHiding = true;
+        await this.hideEl( this.els.container, isSlow )
+        this.isHiding = false
+    }
+
+    hideSync() { this.els.container.classList.add('hidden'); return this }
+
+    _hideEl( el, resolve, hash, isSlow ) {
+        el.removeEventListener( 'animationend', this[ hash ] )
+        el.classList.add('hidden')
+        el.classList.remove(`animate-out${ isSlow ? '-slow' : ''}`)
+        delete this[hash]
+        resolve()
+    }
+
+    hideEl( el, isSlow ) {
+        if( this.isHidden( el ) ) return
+
+        const uuid = this.UUID(),
+            hash = `${uuid}Hide`
+
+        return new Promise( resolve => {
+            this[ hash ] = e => this._hideEl( el, resolve, hash, isSlow )
+            el.addEventListener( 'animationend', this[ hash ] )
+            el.classList.add(`animate-out${ isSlow ? '-slow' : ''}`)
+        } )
+    }
+
+    initialize() {
+        return { ...this, els: { }, views: { } }
+    }
+
+    isAllowed( user ) {
+        if( !this.requiresRole ) return true
+            
+        const userRoles = new Set( user.data.roles )
+
+        if( typeof this.requiresRole === 'string' ) return userRoles.has( this.requiresRole )
+
+        if( Array.isArray( this.requiresRole ) ) {
+            const result = this.requiresRole.find( role => userRoles.has( role ) )
+
+            return result !== undefined
+        }
+
+        return false
+    }
+    
+    isHidden( el ) { return el ? el.classList.contains('hidden') : this.els.container.classList.contains('hidden') }
+
+    onLogin() {
+
+        if( !this.isAllowed( this.user ) ) return this.scootAway()
+
+        this.initialize().render()
+    }
+
+    async onNavigation() {
+        await this.show()
+    }
```

Comment:
You don't really need the async/await here when you're just returning a single statement promise

```
onNavigation() {
    return this.show()
}
```